src/main/java/de/tum/in/www1/artemis/domain/quiz/QuizExercise.java

```diff
@@ -458,7 +458,9 @@ public Result findLatestRatedResultWithCompletionDate(Participation participatio
             // only rated results are considered relevant
             Result latestRatedResult = null;
             for (Result result : participation.getResults()) {
-                if (result.isRated() == Boolean.TRUE && (latestRatedResult == null || latestRatedResult.getCompletionDate().isBefore(result.getCompletionDate()))) {
+                Boolean isAssessmentOver = assessmentDueDate == null || assessmentDueDate.isBefore(ZonedDateTime.now());
```

Comment:
A couple of gripes:
- `isAssessmentOver` is calculated in every loop iteration
- The same issue as described above with the parameter `assessmentDueDate`

Reply:
good point i moved it